### DISCUSSION TITLE
Make UTM Remote able to connect to standard Libvirt over TCP or SSH

### DISCUSSION THREAD

**Discussion by irusensei** [utmapp/UTM#6673] [Ideas]
Hi. This is somewhat related to https://github.com/utmapp/UTM/discussions/4630 which didn't seem to have got a lot of visibility.

Basically, it would be great to either have the UTM Remote App be able to connect to a Linux server running Libvirtd on Linux (or anything Libvirt in general). Reason behind it is that sometimes you need some good old x86 cruft to run as native as possible. KVM virtual machines also have a lot more capabilities than what's available on MacOS hypervisor.

As of today, there is no proper implementation of virt-manager to iOS and MacOS. Virt-manager runs but it's very janky and the very few remote clients seems to be very niche (just spice, just proxmox) don't seem to be actively maintained.  UTM Remote seems like a great way to fill that gap. Of course I wouldn't expecting the VMs to be fully configurable through UTM remote.

I'm totally fine having this feature being locked behind a paid purchase. 

Thank you.

**Comment by tangowithfoxtrot** [comment]
Similarly, it'd be nice if the UTM Remote functionality could either be integrated into the macOS UTM app, or the UTM Remote app could be published for macOS.

I have another Mac running UTM Server that I can't connect to because there's no option to connect to a remote server in the macOS app and sideloading UTM Remote on a Mac just crashes when you try to connect.

In any case, enabling macOS users to connect to remote servers would also fill in the gaps for the lack of virt-manager on macOS as well.
